anni-playback/src/sources/cached_http/mod.rs

```diff
@@ -66,7 +66,7 @@ impl CachedHttpSource {
         let is_buffering = Arc::new(AtomicBool::new(true));
         let pos = Arc::new(AtomicUsize::new(0));
 
-        let (url, duration) = url().ok_or(anyhow!("no audio"))?;
+        let (url, duration) = url().ok_or(OpenTrackError::NoAvailableAnnil)?;
```

Comment:
maybe url should return a `Result` instead of a `Option`?

Reply:
The `url` parameter is intended for providing the url lazily, which, in our case, iterates over a list of annils and returns the first one without any error. Errors generated in the process are 1) difficult to report, 2) not important since we just need at least one annil backend to work and not care about some of them's failure. So I choose to log instead of returning them.